apps/studio/components/grid/components/header/Header.tsx

```diff
@@ -327,6 +328,37 @@ const RowHeader = ({ table, sorts, filters }: RowHeaderProps) => {
     setIsExporting(false)
   }
 
+  async function onRowsExportSQL() {
+    setIsExporting(true)
+
+    if (allRowsSelected && totalRows > MAX_EXPORT_ROW_COUNT) {
+      toast.error(
+        `Sorry! We're unable to support exporting of SQL for row counts larger than ${MAX_EXPORT_ROW_COUNT.toLocaleString()} at the moment.`
```

Comment:
Worth giving the user a workaround here? Linking to a guide on how to use pg_dump or something... https://supabase.com/docs/reference/cli/supabase-db-dump

Reply:
good point!

Reply:
and also i realised this validation is not in EntityListItem either - will fix that

Reply:
okay pushed up the fixes ^ will look like this (ignore the 5, i made the limit smaller to trigger the toast)
<img width="397" alt="image" src="https://github.com/user-attachments/assets/588ddf26-e4c3-430d-b603-ed23b5dd7ece">

just FYI though i don't think the toast here communicates clearly RE the docs, but we'll update it once this PR https://github.com/supabase/supabase/pull/28250 goes through, cause I think using sonner will be better, but I don't wanna introduce a 3rd toast in this PR just yet